capsules/src/net/ipv6/ip_utils.rs

```diff
@@ -115,7 +115,7 @@ impl IPAddr {
 
 pub fn compute_udp_checksum(
     ip6_header: &IP6Header,
-    udp_header: &UDPHeader,
+    udp_header: UDPHeader,
```

Comment:
I wish I understood clearly why this doesn't also apply to the `ip6_header` -- the difference I see in their use is that `udp_header` only has function calls, while `ip6_header` accesses fields. My take then is that if this function ever changed to reference any fields in the `udp_header` directly, then passing by reference would be "necessary" to still be efficient :/

Reply:
I think `IP6Header` is too big fit in the registers ( > 16 bytes). IIRC function arguments are passed in r0-r3. 

However, the size limit seems to be configurable via `trivial_copy_size_limit`. I don't know how good `clippy` is to handle different platforms though...